pkg/operation/botanist/component/logging/kube_rbac_proxy.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
+
+import (
+	"context"
+	"errors"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/common"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ShootNodeLoggingManagedResourceName is the name of managed resources associated with Loki's kube-rbac-proxy and Promtail's RBAC.
+	ShootNodeLoggingManagedResourceName = "shoot-node-logging"
+	// KubeRBACProxyImageName is the name of the kube-rbac-proxy image.
+	KubeRBACProxyImageName = common.LokiKubeRBACProxyName
+	// KubeRBACProxyUserName is the name of the user used by Kube-RBAC-Proxy to make delegating authorization decisions.
+	KubeRBACProxyUserName = "gardener.cloud:logging:kube-rbac-proxy"
+	// PromtailRBACName is the name of the user used by promtail to auth gains Kube-RBAC-Proxy
+	PromtailRBACName = "gardener.cloud:logging:promtail"
+)
+
+// KubeRBACProxyOptions are the options for the kube-rbac-proxy.
+type KubeRBACProxyOptions struct {
+	// Client to create resources with.
+	Client client.Client
+	// Namespace in the seed cluster.
+	Namespace string
+	// IsShootNodeLoggingActivated flag enables or disables the shoot node logging
+	IsShootNodeLoggingActivated bool
+}
+
+// NewKubeRBACProxy creates a new instance of kubeRBACProxy for the kube-rbac-proxy.
+func NewKubeRBACProxy(so *KubeRBACProxyOptions) (component.DeployWaiter, error) {
```

Comment:
It seems `component.Deployer` is sufficient?

Reply:
Yes it is